### PR TITLE
New: Add script to upgrade box-annotations version in Preview

### PR DESCRIPTION
**Pull request by pramodsum**

### CODE REVIEW

File: build/upgrade_annotations.sh [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        echo "----------------------------------------------------------------------"
+    else
+        echo "----------------------------------------------------------------------"
+        echo "Error: Failed to upgrade to $CURRENT_VERSION"
+        echo "----------------------------------------------------------------------"
+    fi
+}
+
+get_changelog() {
+    CHANGELOG="$(curl https://raw.githubusercontent.com/box/box-annotations/master/CHANGELOG.md | awk '/\#/{f=1} f{print; if (/\<a/) exit}' | grep . | sed 's/\<a.*//')";
```
Comment (pramodsum) [inline build/upgrade_annotations.sh]: Regex breakdown
```
# Get raw changelog from github
curl https://raw.githubusercontent.com/box/box-annotations/master/CHANGELOG.md 

# Get latest commit by parsing raw changelog for first instance of # until the next instance of <a> tag 
awk '/\#/{f=1} f{print; if (/\<a/) exit}' 

# Remove extra new lines
grep . 

# Removes ending <a> tag from string
sed 's/\<a.*//'
```

File: build/upgrade_annotations.sh [hunk 2]
```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+export NODE_PATH=$NODE_PATH:./node_modules
+
+update_version() {
+    CURRENT_VERSION="$(cat package.json | grep box-annotations | sed 's/.*\^\(.*\)\".*/\1/')";
```
Comment (pramodsum) [inline build/upgrade_annotations.sh]:
```
# Gets string between ^ and " characters from line containing box-annotations in package.json 
# i.e. "box-annotations": "^0.1.0",
sed 's/.*\^\(.*\)\".*/\1/'
```

Comment (jeremypress) [reply build/upgrade_annotations.sh]: what does the 1 at the end mean?

File: build/upgrade_annotations.sh [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+get_changelog() {
+    CHANGELOG="$(curl https://raw.githubusercontent.com/box/box-annotations/master/CHANGELOG.md | awk '/\#/{f=1} f{print; if (/\<a/) exit}' | grep . | sed 's/\<a.*//')";
+}
+
+
+push_to_github() {
+    # Add new files
+    git add . && git commit -m "Update: box-annotations to v$UPDATED_VERSION" -m "https://github.com/box/box-annotations/releases" -m "$CHANGELOG"
```
Comment (jeremypress) [inline build/upgrade_annotations.sh]: nice!

**Pull request merged**